packages/core/node/core-node-server-internal/src/node_service.ts

```diff
@@ -66,8 +68,9 @@ export class NodeService {
     loggingSystem.setGlobalContext({ service: { node: { roles } } });
     this.log.info(`Kibana process configured with roles: [${roles.join(', ')}]`);
 
-    this.roles = NODE_ACCEPTED_ROLES.reduce((acc, curr) => {
-      return { ...acc, [camelCase(curr)]: roles.includes(curr) };
+    // We assume the combination of node roles has been validated and avoid doing additional checks here.
+    this.roles = NODE_ALL_ROLES.reduce((acc, curr) => {
+      return { ...acc, [camelCase(curr)]: (roles as string[]).includes(curr) };
```

Comment:
I'm guessing the casting is necessary here due to the change below to have `getNodeRoles` return `NodeRolesConfig`?

Reply:
I actually think this might be a TS issue 😅 . Using `.includes` here maps the `searchElement` to `never`. I think this is because of the slightly weird type union...

<img width="827" alt="Screenshot 2023-02-24 at 14 08 47" src="https://user-images.githubusercontent.com/8155004/221186472-097ad375-34b0-41ed-9cc6-4966c48f2be6.png">

Reply:
Ah, OK this looks like the "expected" TS behaviour regarding readonly (or `as const`ed) array types.

https://stackoverflow.com/a/69368653

For our purposes casting to `string[]` seems to most closely resemble how we want to treat the array here.